cdap-app-fabric/src/test/java/io/cdap/cdap/internal/tethering/TetheringClientHandlerTest.java

```diff
@@ -240,61 +243,78 @@ public void setUp() throws Exception {
 
   @After
   public void tearDown() throws Exception {
+    deleteTetheringIfNeeded(SERVER_INSTANCE);
     Assert.assertEquals(Service.State.TERMINATED, tetheringAgentService.stopAndWait());
+  }
 
+  @Test
+  public void testTetheringCreationFails() throws IOException {
```

Comment:
Just a note for future reference (not worth refactoring in this PR).

Much of the existing codebase doesn't follow this pattern, but good practice is to have a http handler, a service, and a store, (and a client where applicable) with unit tests for each layer. Most of the actual logic should be unit tested at the service layer and not at the http handler layer.

Reply:
Noted, thanks.